### DISCUSSION TITLE
How to set cookie in custom server nextjs

### DISCUSSION THREAD

**Discussion by akshaymhk** [vercel/next.js#31997] [Help]
I want to set cookie server side, But unable to achieve this. 
My custom server looks like below, Please suggest proper way to set cookie from server.

`
app.prepare().then(() => {
  createServer((req, res) => {
    // Be sure to pass `true` as the second argument to `url.parse`.
    // This tells it to parse the query portion of the URL.
    const parsedUrl = parse(req.url, true)
    const { pathname, query } = parsedUrl

    if (pathname === '/a') {
      app.render(req, res, '/a', query)
    } else if (pathname === '/b') {
      app.render(req, res, '/b', query)
    } else {
      handle(req, res, parsedUrl)
    }
  }).listen(3000, (err) => {
    if (err) throw err
    console.log('> Ready on http://localhost:3000')
  })
})
`

**Comment by jamesmosier** [comment]
I couldn't find a way to do this with Next & a default custom Node server. 

Typically with a Node HTTP server, you'd set a cookie like this:

```js
  response.writeHead(200, {
    'Set-Cookie': 'your_cookie=123',
    'Content-Type': 'text/html'
  });
  response.end();
```

but in this case Next/Node are internally handling the response therefore the above does not work.

If you need to use a custom server then I would recommend using a [custom Express server](https://github.com/vercel/next.js/tree/canary/examples/custom-server-express) as this is easier to work with cookies in my opinion. There is a module for Express called [cookie-parser](http://expressjs.com/en/resources/middleware/cookie-parser.html) that you could use to get/set cookies with.
